freud/common.pyx

```diff
@@ -125,6 +133,70 @@ cdef class Compute:
         return wrapper
 
 
+cdef class PairCompute(Compute):
+    R"""Parent class for all compute classes in freud that depend on finding
+    nearest neighbors.
+
+    The purpose of this class is to consolidate some of the logic for parsing
+    the numerous possible inputs to the compute calls of such classes. In
+    particular, this class contains a helper function that calls the necessary
+    functions to create NeighborQuery and NeighborList classes as needed, as
+    well as dealing with boxes and query arguments.
+
+    .. moduleauthor:: Vyas Ramasubramani <vramasub@umich.edu>
+    """
+    def preprocess_arguments(self, box, points, query_points=None, nlist=None,
+                              query_args=None):
+        """Process standard compute arguments into freud's internal types by
+        calling all the required internal functions.
+
+        The preprocessing of boxes and points into NeighborQuery objects, the
```

Comment:
Reword: "This function handles preprocessing..." instead of "The preprocessing... are handled by this function."

Reply:
Please add `:py:class:` designations for classes and write argument names like `query_points` in code font.